zeebe/docs/generalized_distribution.md

```diff
@@ -66,17 +66,27 @@ metadata and raw record value are removed from the state.
 The diagram shows the flow of commands and events. What it doesn't highlight is the record that
 belongs to these.
 
-The `CommandDistributionRecord` is relatively simple. It contains 4 properties which are required to
+The `CommandDistributionRecord` is relatively simple. It contains 6 properties which are required to
 distribute a command successfully.
 
 #### Partition id
 
 The value of the partition id depends on the `CommandDistributionIntent`:
 
 - `STARTED` and `FINISHED`: The partition id is equal to the distributing partition
-- `DISTRIBUTING`, `ACKNOWLEDGE` and `ACKNOWLEDGED`: The partition id is equal to the partition that
+- `ENQUEUED`, `DISTRIBUTING`, `ACKNOWLEDGE` and `ACKNOWLEDGED`: The partition id is equal to the partition that
   the command is distributed to.
 
+#### Queue id
+
+This property is optional (empty string indicates no queue id). It is set on the `ENQUEUED` and `STARTED` intents.
+If it is set, the next property `queueInsertionKey` must be set as well.
+
+#### Queue insertion key
+
+This property is optional (-1 indicates no queue insertion key) unless the `queueId` is set. It is set on the `ENQUEUED` and `STARTED` intents.
+This indicates the position in the queue where this command is inserted.
```

Comment:
🔧 Don't forget to remove this when you remove the insertion key from the record 🙂